lib/uksparsebuf/include/uk/sparsebuf/util.h

```diff
@@ -0,0 +1,124 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2025, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+/* Sparse buffer utilities for API users.
+ *
+ * This covers common non-trivial functions that operate on byte granularity.
+ *
+ * Because these functions call into state-modifying functions of the API, they
+ * are similarly not thread safe and callers must ensure mutual exclusion.
+ */
+
+#ifndef __UK_SPARSEBUF_UTIL_H__
+#define __UK_SPARSEBUF_UTIL_H__
+
+#include <uk/sparsebuf.h>
+
+/**
+ * Zero out `len` bytes starting at sparse buffer offset `off`, using `cur`.
+ */
+static inline
+void uk_sparsebuf_memclear(struct uk_sparsebuf_cur *cur, __sz off, __sz len)
+{
+	char *buf = __NULL;
+	__sz blen = uk_sparsebuf_memat(off, cur, &buf);
```

Comment:
Just a heads up, this `uk_sparsebuf_memat` is also inlined and called in several places (2 more in ukfs-ramfs), just FYI, cus `memclear` is also inlined and called in several places as well (e.g. twice in `uk_sparsebuf_punch_hole`).

Reply:
Nope I am not suggesting anything. Just making an observation in case you lost track of this as you were writing the rest of the code. If you think it's fine then ok, it's not used in that many places anyway.